core/state/trie_prefetcher.libevm.go

```diff
@@ -68,43 +71,66 @@ func (c *prefetcherConfig) applyTo(sf *subfetcher) {
 	}
 }
 
-func (sf *subfetcher) wait() {
-	if w := sf.pool.workers; w != nil {
-		w.Wait()
+func (p *triePrefetcher) abortFetchersAndReleaseWorkerPools() {
+	// Calling abort() sequentially may result in later fetchers accepting new
+	// work in the interim.
+	var wg sync.WaitGroup
+	for _, f := range p.fetchers {
+		wg.Add(1)
+		go func(f *subfetcher) {
+			f.abort()
+			wg.Done()
+		}(f)
+	}
```

Comment:
I don't think this is needed, the user of triePrefetcher should not call `prefetch` after or concurrently with `abort`.

Also this doesn't guarantee the cancellation.

IMO we can remove this part, rename the fn to releaseWorkerPools() and call it after the sequential abort loop before line 109

Reply:
Agreed, it was YAGNI thinking on my part.